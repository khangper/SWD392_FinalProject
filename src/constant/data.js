export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
