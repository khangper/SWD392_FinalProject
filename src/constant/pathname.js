export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
