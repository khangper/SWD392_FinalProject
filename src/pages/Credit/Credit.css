.credit-wallet-icon {
  width: 20px;
  height: 20px;
  position: relative;
  flex-shrink: 0;
}

.credit-icon {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-end;
  padding: 0 0 2.8px;
}

.credit-credit {
  position: relative;
  line-height: 26px;
  font-weight: 550;
  color: #333;
  min-width: 73px;
  font-size: 18px;
  margin-bottom: 0 !important;
}

.credit-label,
.credit-title {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
}

.credit-label {
  align-items: flex-end;
  gap: 10px;
}

.credit-title {
  align-items: flex-start;
}

.credit-next-wrapper,
.credit-account-wrapper {
  border: 1px solid #efefef;
  background-color: #fff;
  max-width: 100%;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
  align-self: stretch;
  border-radius: 3px;
}

.credit-label {
  flex: 1;
  position: relative;
  line-height: 21px;
  font-weight: 550;
  max-width: 100%;
  color: #333;
  font-size: 16px;
}

.credit-label-wrapper,
.credit-via-wrapper,
.credit-added-wrapper {
  align-self: stretch;
  border-bottom: 1px solid #efefef;
  box-sizing: border-box;
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  max-width: 100%;
}

.credit-label-wrapper {
  flex-direction: row;
  padding: 16.8px 28px 16px;
}
.credit-coin {
  position: relative;
  line-height: 24px;
  font-size: 26px;
  display: inline-block;
  color: #a1a1a1;
  white-space: nowrap;
  font-weight: 550;
}

.credit-via {
  position: relative;
  line-height: 24px;
  align-self: stretch;
  font-size: 12px;
  font-weight: 600;
  color: #686f7a;
}

.credit-via-wrapper,
.credit-added-wrapper {
  flex-direction: column;
  padding: 16.8px 20px 16px;
  gap: 10px;
}

.credit-day-wrapper,
.credit-button-wrapper {
  align-self: stretch;
  box-sizing: border-box;
  max-width: 100%;
  padding: 16.8px 20px 16px;
}

.credit-day {
  position: relative;
  line-height: 26px;
  font-size: 12px;
  color: #686f7a;
}

.credit-day span {
  font-weight: 600;
}

.credit-image {
  width: 80px;
  position: relative;
  flex-shrink: 0;
}

.credit-added {
  position: relative;
  line-height: 24px;
  font-size: 14px;
  color: #686f7a;
  font-weight: 550;
}

.credit-added span {
  font-weight: 500;
}

.credit-button, .credit-add-credit-button{
  display: flex;
  padding: 13px 15px;
  justify-content: center;
  align-items: center;
  border-radius: 3px;
  background-color: #ed2a26;
  border: none;
  color: #fff;
  font-weight: bold;
  cursor: pointer;
  width: 100%;
}

.credit-button:hover {
  background-color: #333;
}

.credit-next-account-parent {
  flex: 0.328;
  display: flex;
  flex-direction: column;
  gap: 30px;
}

.credit-parent {
  display: flex;
  flex: 1;
  flex-direction: row;
  gap: 20px;
  align-items: flex-start;
  justify-content: flex-start;
  flex-wrap: wrap;
}

.credit-table-parent {
  flex: 0.672;
}

.credit-wrapper {
  display: flex;
  flex-direction: column;
  gap: 40px;
  padding: 10px;
  flex-wrap: wrap;
}

.credit-input-parent {
  padding: 30px 28px 16px;
  display: flex;
  flex-direction: column;
}

.credit-input-parent label,
.credit-label-input label,
.credit-label-select label {
  position: relative;
  color: #333;
  font-size: 13px;
  line-height: 24px;
  font-weight: 550;
}

.credit-add-balance {
  display: flex;
  flex-direction: column;
  gap: 10px;
  max-width: 100%;
}

.credit-add-credit {
  border: 1px solid #efefef;
  background-color: #fff;
  max-width: 100%;
}

.credit-icon-dollar {
  width: 18px;
  height: 18px;
  color: rgb(172, 172, 172);
}

.credit-button-dollar {
  padding: 8px 5px 5px 18px;
  border: 0;
  background-color: transparent;
}

.credit-icon-dollar:hover {
  color: #333;
}

.credit-icon-input,
.credit-input {
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
}
.credit-input {
  background-color: transparent;
  border: 0;
  outline: 0;
  height: 30px;
  flex-direction: column;
  padding: 5.2px 0 0 15px;
  font-size: 14px;
  max-width: 100%;
  width: 100%;
  color: rgba(0, 0, 0, 0.87);
}

.credit-icon-input {
  flex-direction: row;
  flex: 1;
  border-radius: 3px;
  background-color: #fff;
  border: 1px solid #e5e5e5;
  gap: 1.4px;
  max-width: 100%;
  width: 100%;
}

.credit-icon-input:hover {
  color: #333;
}

.credit-input::placeholder {
  color: lightgray;
}

.credit-input-parent {
  display: flex;
  flex-direction: column;
  gap: 30px;
}

.credit-label-input, .credit-label-select {
  display: flex;
  flex-direction: column;
  gap: 10px;
  width: 100%;
  max-width: 100%;
}

.credit-label-input input,
.credit-select-month .ant-select-selector {
  width: 100%;
  max-width: 100%;
  min-width: 100%;
  padding: 10px 15px;
  background: #fff;
  border: 1px solid #e5e5e5 !important;
  outline: 0;
  color: rgba(0, 0, 0, 0.87);
  border-radius: 3px !important;
  box-shadow: inset 0px 1px 4px rgba(0, 0, 0, 0.02) !important;
}

.credit-select-month .ant-select-selector {
  font-size: 13px;
  padding: 18px 15px !important;
}

.credit-select-month .ant-select-arrow {
  color: #333 !important;
  padding-top: 10px !important;
}

.credit-holder-card, .credit-expiration-parent{
  display: flex;
  flex-direction: row;
  gap: 20px;
  width: 100%;
  max-width: 100%;
}

.credit-label-input input::placeholder {
  color: lightgray;
}

.credit-button-wrappers{
    align-self: stretch;
  box-sizing: border-box;
  max-width: 100%;
  padding: 20px 0;
}

.credit-table table {
    text-align: left;
    font-size: 14px;
    border: 1px solid #e5e5e5;
    width: 100%;
    color: #333;
    border-collapse: collapse;
    border-radius: 3px;
    min-width: 600px;
}

.credit-table th, .credit-table td{
    border-bottom: 1px solid #e5e5e5;
}

.credit-table th{
    padding: 15px;
    font-weight: 550;
    background-color: #ffecec;
}
.credit-table td{
    padding: 15px 20px;
    background-color: #fff;
}

.credit-table-parent{
    display: flex;
    flex-direction: column;
    gap: 30px;
}

.credit{
    padding-bottom: 30px;
}

@media screen and (max-width: 1050px) {
  .credit-parent {
    display: block;
  }

  .credit-wrapper {
    padding-bottom: 30px;
  }
  .credit-table-parent{
    padding-top: 30px;
  }
}
