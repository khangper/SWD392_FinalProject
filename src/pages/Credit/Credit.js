import React from "react";
import "./Credit.css";
import { LuWallet } from "react-icons/lu";
import { Link } from "react-router-dom";
import { PiCurrencyDollarLight } from "react-icons/pi";
import { Select } from "antd";

const Credit = () => {
  const { Option } = Select;
  return (
    <main className="credit">
      <div className="credit-wrapper">
        <div className="credit-title">
          <div className="credit-label">
            <div className="credit-icon">
              <LuWallet className="credit-wallet-icon" />
            </div>
            <h3 className="credit-credit">Credits</h3>
          </div>
        </div>

        <div className="credit-parent">
          <div className="credit-next-account-parent">
            <div className="credit-next-wrapper">
              <div className="credit-label-wrapper">
                <div className="credit-label">Added Credits</div>
              </div>
              <div className="credit-via-wrapper">
                <div className="credit-coin">$100</div>
                <div className="credit-via">via Mastercard</div>
              </div>
              <div className="credit-day-wrapper">
                <div className="credit-day">
                  Your credits will be using on <span>purchase courses</span>
                </div>
              </div>
            </div>
            <div className="credit-account-wrapper">
              <div className="credit-label-wrapper">
                <div className="credit-label">Saved Cards</div>
              </div>
              <div className="credit-added-wrapper">
                <img
                  src="https://gambolthemes.net/html-items/cursus-new-demo/images/Mastercard.svg"
                  alt=""
                  className="credit-image"
                />
                <div className="credit-added">
                  Added: <span>25 Mar 2020</span>
                </div>
              </div>

              <div className="credit-button-wrapper">
                <Link to="#">
                  <button className="credit-button">Change Card</button>
                </Link>
              </div>
            </div>
          </div>

          <div className="credit-table-parent">
            <div className="credit-add-credit">
              <div className="credit-label-wrapper">
                <div className="credit-label">Add Credit Balance</div>
              </div>
              <div className="credit-input-parent">
                <div className="credit-add-balance">
                  <label htmlFor="addbalance">Add Balance</label>
                  <div className="credit-icon-input">
                    <button className="credit-button-dollar">
                      <PiCurrencyDollarLight className="credit-icon-dollar" />
                    </button>
                    <input
                      type="text"
                      className="credit-input"
                      id="addbalance"
                      name="addbalance"
                      placeholder="0"
                    />
                  </div>
                </div>

                <div className="credit-holder-card">
                  <div className="credit-label-input">
                    <label htmlFor="holdername">Holder Name</label>
                    <input
                      type="text"
                      id="holdername"
                      name="holdername"
                      placeholder="Enter Holder Name"
                    />
                  </div>
                  <div className="credit-label-input">
                    <label htmlFor="cardnumber">Card Number</label>
                    <input
                      type="text"
                      name="cardname"
                      id="cardname"
                      placeholder="Card #"
                    />
                  </div>
                </div>
                <div className="credit-expiration-parent">
                  <div className="credit-label-select">
                    <label htmlFor="">Expiration Month</label>
                    <Select
                      className="credit-select-month"
                      placeholder="Month"
                      style={{ color: "lightgray" }}
                    >
                      <Option value="jan">January</Option>
                      <Option value="feb">February</Option>
                      <Option value="mar">March</Option>
                      <Option value="apr">April</Option>
                      <Option value="may">May</Option>
                      <Option value="jun">June</Option>
                      <Option value="jul">July</Option>
                      <Option value="aug">August</Option>
                      <Option value="sep">September</Option>
                      <Option value="oct">October</Option>
                      <Option value="nov">November</Option>
                      <Option value="dec">December</Option>
                    </Select>
                  </div>
                  <div className="credit-label-input">
                    <label htmlFor="year">Expiration Year</label>
                    <input
                      type="text"
                      name="year"
                      id="year"
                      placeholder="Year"
                    />
                  </div>
                  <div className="credit-label-input">
                    <label htmlFor="cvc">Expiration Year</label>
                    <input type="text" name="cvc" id="cvc" placeholder="CVC" />
                  </div>
                </div>

                <div className="credit-button-wrappers">
                  <button className="credit-add-credit-button" type="submit">
                    Add Credit
                  </button>
                </div>
              </div>
            </div>

            <div className="credit-table">
              <table>
                <thead>
                  <tr>
                    <th>Amount</th>
                    <th>Payout Method</th>
                    <th>Date Processed</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td>$50.50</td>
                    <td>Mastercard</td>
                    <td>15 Mar 2020</td>
                  </tr>

                  <tr>
                    <td>$50.50</td>
                    <td>Mastercard</td>
                    <td>15 Feb 2020</td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </main>
  );
};

export default Credit;
