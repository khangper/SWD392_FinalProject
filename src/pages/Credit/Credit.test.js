import React from "react";
import { render, screen } from "@testing-library/react";
import Credit from "../Credit/Credit";
import "@testing-library/jest-dom/extend-expect";
import { renderWithProviders } from "../../constant/test-util";

describe("Credit", () => {
  beforeAll(() => {
    jest.spyOn(console, "error").mockImplementation(() => {});
    jest.spyOn(console, "log").mockImplementation(() => {});
  });

  afterAll(() => {
    console.error.mockRestore();
    console.log.mockRestore();
  });
  it("renders the Credit component", async () => {
    const { container } = renderWithProviders(<Credit />);
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
