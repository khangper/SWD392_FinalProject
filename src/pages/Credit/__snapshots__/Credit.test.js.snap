// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Credit renders the Credit component 1`] = `
<div>
  <main
    class="credit"
  >
    <div
      class="credit-wrapper"
    >
      <div
        class="credit-title"
      >
        <div
          class="credit-label"
        >
          <div
            class="credit-icon"
          >
            <svg
              class="credit-wallet-icon"
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21 12V7H5a2 2 0 0 1 0-4h14v4"
              />
              <path
                d="M3 5v14a2 2 0 0 0 2 2h16v-5"
              />
              <path
                d="M18 12a2 2 0 0 0 0 4h4v-4Z"
              />
            </svg>
          </div>
          <h3
            class="credit-credit"
          >
            Credits
          </h3>
        </div>
      </div>
      <div
        class="credit-parent"
      >
        <div
          class="credit-next-account-parent"
        >
          <div
            class="credit-next-wrapper"
          >
            <div
              class="credit-label-wrapper"
            >
              <div
                class="credit-label"
              >
                Added Credits
              </div>
            </div>
            <div
              class="credit-via-wrapper"
            >
              <div
                class="credit-coin"
              >
                $100
              </div>
              <div
                class="credit-via"
              >
                via Mastercard
              </div>
            </div>
            <div
              class="credit-day-wrapper"
            >
              <div
                class="credit-day"
              >
                Your credits will be using on 
                <span>
                  purchase courses
                </span>
              </div>
            </div>
          </div>
          <div
            class="credit-account-wrapper"
          >
            <div
              class="credit-label-wrapper"
            >
              <div
                class="credit-label"
              >
                Saved Cards
              </div>
            </div>
            <div
              class="credit-added-wrapper"
            >
              <img
                alt=""
                class="credit-image"
                src="https://gambolthemes.net/html-items/cursus-new-demo/images/Mastercard.svg"
              />
              <div
                class="credit-added"
              >
                Added: 
                <span>
                  25 Mar 2020
                </span>
              </div>
            </div>
            <div
              class="credit-button-wrapper"
            >
              <a
                href="/"
              >
                <button
                  class="credit-button"
                >
                  Change Card
                </button>
              </a>
            </div>
          </div>
        </div>
        <div
          class="credit-table-parent"
        >
          <div
            class="credit-add-credit"
          >
            <div
              class="credit-label-wrapper"
            >
              <div
                class="credit-label"
              >
                Add Credit Balance
              </div>
            </div>
            <div
              class="credit-input-parent"
            >
              <div
                class="credit-add-balance"
              >
                <label
                  for="addbalance"
                >
                  Add Balance
                </label>
                <div
                  class="credit-icon-input"
                >
                  <button
                    class="credit-button-dollar"
                  >
                    <svg
                      class="credit-icon-dollar"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 256 256"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M152,122H134V54h10a34,34,0,0,1,34,34,6,6,0,0,0,12,0,46.06,46.06,0,0,0-46-46H134V24a6,6,0,0,0-12,0V42H112a46,46,0,0,0,0,92h10v68H104a34,34,0,0,1-34-34,6,6,0,0,0-12,0,46.06,46.06,0,0,0,46,46h18v18a6,6,0,0,0,12,0V214h18a46,46,0,0,0,0-92Zm-40,0a34,34,0,0,1,0-68h10v68Zm40,80H134V134h18a34,34,0,0,1,0,68Z"
                      />
                    </svg>
                  </button>
                  <input
                    class="credit-input"
                    id="addbalance"
                    name="addbalance"
                    placeholder="0"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="credit-holder-card"
              >
                <div
                  class="credit-label-input"
                >
                  <label
                    for="holdername"
                  >
                    Holder Name
                  </label>
                  <input
                    id="holdername"
                    name="holdername"
                    placeholder="Enter Holder Name"
                    type="text"
                  />
                </div>
                <div
                  class="credit-label-input"
                >
                  <label
                    for="cardnumber"
                  >
                    Card Number
                  </label>
                  <input
                    id="cardname"
                    name="cardname"
                    placeholder="Card #"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="credit-expiration-parent"
              >
                <div
                  class="credit-label-select"
                >
                  <label
                    for=""
                  >
                    Expiration Month
                  </label>
                  <div
                    class="ant-select ant-select-outlined credit-select-month css-dev-only-do-not-override-f7vrd6 ant-select-single ant-select-show-arrow"
                    style="color: lightgray;"
                  >
                    <div
                      class="ant-select-selector"
                    >
                      <span
                        class="ant-select-selection-search"
                      >
                        <input
                          aria-autocomplete="list"
                          aria-controls="rc_select_TEST_OR_SSR_list"
                          aria-expanded="false"
                          aria-haspopup="listbox"
                          aria-owns="rc_select_TEST_OR_SSR_list"
                          autocomplete="off"
                          class="ant-select-selection-search-input"
                          id="rc_select_TEST_OR_SSR"
                          readonly=""
                          role="combobox"
                          style="opacity: 0;"
                          type="search"
                          unselectable="on"
                          value=""
                        />
                      </span>
                      <span
                        class="ant-select-selection-placeholder"
                      >
                        Month
                      </span>
                    </div>
                    <span
                      aria-hidden="true"
                      class="ant-select-arrow"
                      style="user-select: none;"
                      unselectable="on"
                    >
                      <span
                        aria-label="down"
                        class="anticon anticon-down ant-select-suffix"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="down"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </div>
                <div
                  class="credit-label-input"
                >
                  <label
                    for="year"
                  >
                    Expiration Year
                  </label>
                  <input
                    id="year"
                    name="year"
                    placeholder="Year"
                    type="text"
                  />
                </div>
                <div
                  class="credit-label-input"
                >
                  <label
                    for="cvc"
                  >
                    Expiration Year
                  </label>
                  <input
                    id="cvc"
                    name="cvc"
                    placeholder="CVC"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="credit-button-wrappers"
              >
                <button
                  class="credit-add-credit-button"
                  type="submit"
                >
                  Add Credit
                </button>
              </div>
            </div>
          </div>
          <div
            class="credit-table"
          >
            <table>
              <thead>
                <tr>
                  <th>
                    Amount
                  </th>
                  <th>
                    Payout Method
                  </th>
                  <th>
                    Date Processed
                  </th>
                </tr>
              </thead>
              <tbody />
            </table>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
